src/shared_key.rs

```diff
@@ -22,9 +23,85 @@ pub enum SharedKeyError {
     Generate,
 }
 
+const DITTO_IDENTITY_TAG: &str = "DITTO IDENTITY";
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+struct X509Auth {
+    certificates: Vec<Vec<u8>>,
+    private_key: Vec<u8>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+enum JwtAuth {
+    AuthUrl { url: String },
+    AuthProvider { name: String },
+    Token { token: String },
```

Comment:
The options are more like:

```suggestion
enum JwtAuth {
    OnlineAuth { url: String, provider: String, token_credential: String },
    Static { jwt: String },
```